_includes/code/animation/basic_staggered_animation/main.dart

```diff
@@ -0,0 +1,204 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that
+// can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart' show timeDilation;
+
+class StaggerAnimation extends StatelessWidget {
+  StaggerAnimation({ Key key, this.controller }) :
+
+    // Each animation defined here transforms its value between the subset of
+    // the duration defined by the animation's interval. Most of the Intervals
+    // cause the animations' transforms to be "staggered", i.e. to start one
+    // after another.
+
+    opacity = new Tween<double>(
+      begin: 0.0,
+      end: 1.0,
+    ).animate(
+      new CurvedAnimation(
+        parent: controller,
+        curve: new Interval(
+          0.0, 0.100,
+          curve: Curves.ease,
+        ),
+      ),
+    ),
+
+    width = new Tween<double>(
+      begin: 50.0,
+      end: 150.0,
+    ).animate(
+      new CurvedAnimation(
+        parent: controller,
+        curve: new Interval(
+          0.125, 0.250,
+          curve: Curves.ease,
+        ),
+      ),
+    ),
+
+    height = new Tween<double>(
+      begin: 50.0,
+      end: 150.0
+    ).animate(
+      new CurvedAnimation(
+        parent: controller,
+        curve: new Interval(
+          0.250, 0.375,
+          curve: Curves.ease,
+        ),
+      ),
+    ),
+
+    padding = new EdgeInsetsTween(
+      begin: const EdgeInsets.only(bottom: 16.0),
+      end: const EdgeInsets.only(bottom: 75.0),
+    ).animate(
+      new CurvedAnimation(
+        parent: controller,
+        curve: new Interval(
+          0.250, 0.375,
+          curve: Curves.ease,
+        ),
+      ),
+    ),
+
+    borderRadius = new BorderRadiusTween(
+      begin: new BorderRadius.circular(4.0),
+      end: new BorderRadius.circular(75.0),
+    ).animate(
+      new CurvedAnimation(
+        parent: controller,
+        curve: new Interval(
+          0.375, 0.500,
+          curve: Curves.ease,
+        ),
+      ),
+    ),
+
+    color = new ColorTween(
+      begin: Colors.indigo[100],
+      end: Colors.orange[400],
+    ).animate(
+      new CurvedAnimation(
+        parent: controller,
+        curve: new Interval(
+          0.500, 0.750,
+          curve: Curves.ease,
+        ),
+      ),
+    ),
+
+    super(key: key);
+
+  final Animation<double> controller;
+  final Animation<double> opacity;
+  final Animation<double> width;
+  final Animation<double> height;
+  final Animation<EdgeInsets> padding;
+  final Animation<BorderRadius> borderRadius;
+  final Animation<Color> color;
+
+  Widget _buildAnimation(BuildContext context, Widget child) {
```

Comment:
It might be worth putting a comment here.

```
  // This function is called each the controller "ticks" a new frame.
  // When it runs, all of the animation's values will have been
  // updated to reflect the controller's current value.
```